docs/modules/ROOT/pages/api/governance.adoc

```diff
@@ -667,3 +671,188 @@ Delegates votes from the sender to `delegatee`.
 ==== `[.contract-item-name]#++delegate_by_sig++#++(delegator: ContractAddress, delegatee: ContractAddress, nonce: felt252, expiry: u64, signature: Array<felt252>)++` [.item-kind]#external#
 
 Delegates votes from `delegator` to `delegatee`.
+
+
+[.contract]
+[[VotesComponent]]
+=== `++VotesComponent++` link:https://github.com/OpenZeppelin/cairo-contracts/blob/release-v0.17.0/packages/governance/src/votes/votes.cairo[{github-icon},role=heading-link]
+
+[.hljs-theme-dark]
+```cairo
+use openzeppelin_governance::votes::VotesComponent;
+```
+
+By default, token balance does not account for voting power. This makes transfers cheaper. The downside is that it requires users to delegate to themselves in order to activate checkpoints and have their voting power tracked.
+
+NOTE: When using this module, your contract must implement the {VotingUnitsTrait}. For convinience, this is done automatically for `ERC20` and `ERC721` tokens.
+
+[.contract-index#VotesComponent-Embeddable-Impls]
+.Embeddable Implementations
+--
+[.sub-index#VotesComponent-Embeddable-Impls-VotesImpl]
+.VotesImpl
+* xref:#VotesComponent-get_votes[`++get_votes(self, account)++`]
+* xref:#VotesComponent-get_past_votes[`++get_past_votes(self, account, timepoint)++`]
+* xref:#VotesComponent-get_past_total_supply[`++get_past_total_supply(self, timepoint)++`]
+* xref:#VotesComponent-delegates[`++delegates(self, account)++`]
+* xref:#VotesComponent-delegate[`++delegate(self, delegatee)++`]
+* xref:#VotesComponent-delegate_by_sig[`++delegate_by_sig(self, delegator, delegatee, nonce, expiry, signature)++`]
+--
+
+[.contract-index]
+.Internal implementations
```

Comment:
We should add a `VotingUnitsTrait implementations` section so users are aware of the implementations provided by default with the component, and also can see the comments explaining the formula followed for internal accounting.

Reply:
I added it and I'm not sure I love the result. Can you take a look and let me know if this is what you had in mind? Happy to iterate over this if needed.

Reply:
Some notes:

- The `.VotingUnitsTrait implementations` should be the first section before `Embeddable implementations` for consistency with other pages having custom extra items.
The NOTE splitting the sections should not appear here but in the section expanding the function descriptions below (which we still need to add).
- We need to add the section below with the expanded descriptions (same comment that the function has in the code) and should add that they will work out-of-box if the corresponding component is implemented.

Reply:
+1. I vote for `VOTING UNITS TRAIT IMPLEMENTATIONS`.

Reply:
Sorry, really struggled to understand what you meant here. So I followed an example from another doc(vesting wallet in this case) and went ahead with that pattern, where we have the trait implementation section before the Embeddable Implementations and we just link to the trait section. I think this makes it clear, since that section includes a more detailed description and even a WARNING, but feel free to edit and commit if you feel something is missing.

Reply:
VestingWallet is a preset which is not documented in the same way. The updates you did are great, the only thing missing is a section in the component with the functions explained (copy pasted the impl function comment), and the header section linking to that instead.

Reply:
Currently the first section below the header is `[#VotesComponent-Functions]`, we need to add a [#VotesComponent-VotingUnitsTrait-Implementations] above that, with the two impls names as sections, each of them with the function as sub-item and the comment as descriptor.